compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -749,6 +749,50 @@ fn infer_placeholder_type(
     span: Span,
     item_ident: Ident,
 ) -> Ty<'_> {
+    fn contains_anonymous(ty: Ty<'_>) -> bool {
```

Comment:
So, you could instead make this fold through the type and make it `nameable_type`. If the type can be named (or can be made nameable by replacing the fndef with a fnptr), then it returns `Some(Ty)`, otherwise, it would return `None`.

Reply:
I have now dropped the function and just called `fold_ty()` directly.